runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3174,6 +3189,68 @@ remoteCompile(
          JITServer::ServerMemoryState nextMemoryState = std::get<9>(recv);
          updateCompThreadActivationPolicy(compInfoPT, nextMemoryState);
          }
+      else if (JITServer::MessageType::jitDumpPrintIL == response)
+         {
+         // A server compilation thread has crashed, and the crashing thread
+         // is generating a JitDump. Use the existing compilation to print IL
+         // of the method causing the crash and then schedule a JitDump recompilation
+         TR::CompilationInfoPerThreadBase::UninterruptibleOperation uop(*compInfoPT);
+         releaseVMAccess(vmThread);
+         while(!handleServerMessage(client, compiler->fej9vm(), response));
+         acquireVMAccessNoSuspend(vmThread);
+         
+         TR_ASSERT_FATAL(
+            response == JITServer::MessageType::compilationThreadCrashed
```

Comment:
If the message we receive is `compilationFailure` below we should not read the FILE sent by the server with a `compilationThreadCrashed` message.
Also, if we receive a message type that triggers this asserts, maybe we should print the bad message type in this assert. This way, if the assert ever triggers, we have more information about what went wrong.